### DISCUSSION TITLE
Is the Live Crystal Asteroid meant to have such a low frame rate?

### DISCUSSION THREAD

**Discussion by opusforlife2** [endless-sky/endless-sky#10626] [Lore]
When zooming in to it fully, you can see it has a frame rate that makes it look like the graphics are stuttering. Is this intentional?

**Comment by Saugia** [comment]
It's not so much intentional as it is a limitation of the game's graphics. We don't want to add more frames than what it currently has because that amount of images adds up very quickly for users on lower-end systems. Currently, we're waiting for something like PNG sequencing or other ways to have animated sprites take less space.

**Reply by warp-core** [reply]
The asteroids actually aren't *that* big.
In terms of file sizes, the Astral Cetacean and Ember Waste Node are larger than all the asteroids combined.
The runtime memory required is considerably larger, with the live crystal asteroids (having the most pixels of any asteroids) being about 20 MB (all asteroids combined at less than 60 MB, meaning this asteroid actually has more than half as many pixels as every other asteroid combined). That's considerable, but increasing it, if necessary, would not be unthinkable.

The "livecrystal" does have one of the highest frame counts of any asteroid, which may indicate that it was designed to be rendered at a higher framerate than it currently is. They are attributed to @GefullteTaubenbrust2, who may be able to offer more insight there.

If, with so many frames, it does indeed still not look right, it may be worth reconsidering the design of the sprites. Perhaps a different rotation that allows fewer frames to create a smooth loop.

**Reply by Saugia** [reply]
I've talked about adding large asteroids to Gegno space too, and after some tests, for the rotation to look good, it would have to be about twice what the livecrystal is, or around 120 or so per asteroid. Because the rotation is on two axis, it requires more frames to look better, but the more efficient option would be to simply have a single plane rotation from top view, just as ships. Though I'm not very fond of that option. Rendering at higher frame rates, especially for larger objects, looks rather silly with the speed-size ratio.

If we do want it to look better, and are willing to accept the higher frame counts, I'd be fine with having it increased.

**Reply by opusforlife2** [reply]
To be honest, it could just be accepted in-lore as an oddity that presents itself with this asteroid for some reason due to the physics created by the dimension bubble. It's quite wacky to look at.

**Reply by Saugia** [reply]
I wouldn't overcomplicate video game-based limitations with the game's lore. It's simply a graphical reason.

Either way, I hope the above answers your question.
